topostats/utils.py

```diff
@@ -237,5 +237,5 @@ def create_empty_dataframe(columns: set = ALL_STATISTICS_COLUMNS, index: tuple =
         Empty Pandas DataFrame.
     """
     empty_df = pd.DataFrame(columns=columns)
-    empty_df.index.names = [index]
+    empty_df = empty_df.set_index(index)
```

Comment:
I'm surprised the difference between these two has such an impact, I naively thought they were synonyms.

Reply:
Ah, I now realise these aren't synonyms `empty_df.set_index(index)` is saying to make the existing column the index rather than just naming the index with the value of `index`.